cubano-config/src/main/java/org/concordion/cubano/utils/CaselessProperties.java

```diff
@@ -1,30 +1,40 @@
-package org.concordion.cubano.utils;
-
-import java.util.Properties;
-
-/**
- * A {@link Properties} implementation that ignores the case of the properties.
- *
- * @author Andrew Sumner
- */
-public class CaselessProperties extends Properties {
-    private static final long serialVersionUID = 4112578634029874840L;
-
-    @Override
-    public Object put(Object key, Object value) {
-        String lowercase = ((String) key).toLowerCase();
-        return super.put(lowercase, value);
-    }
-
-    @Override
-    public String getProperty(String key) {
-        String lowercase = key.toLowerCase();
-        return super.getProperty(lowercase);
-    }
-
-    @Override
-    public String getProperty(String key, String defaultValue) {
-        String lowercase = key.toLowerCase();
-        return super.getProperty(lowercase, defaultValue);
-    }
-}
+package org.concordion.cubano.utils;
+
+import java.util.Map;
+import java.util.Properties;
+import java.util.TreeMap;
+
+/**
+ * A {@link Properties} implementation that ignores the case of the properties.
+ * 
+ * The keys are stored untouched so that if you need the original case you can still get them 
+ * using <code>Enumeration&lt;String&gt; en = (Enumeration<String>) properties.propertyNames();</code>
+ *
+ * @author Andrew Sumner
+ */
+public class CaselessProperties extends Properties {
+    private static final long serialVersionUID = 4112578634029874840L;
+    
+    Map<String, String> lookup = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+
+    @Override
+    public Object put(Object key, Object value) {
+    		lookup.put(((String) key).toLowerCase(), (String) key);

```

Comment:
The formatting is using a mix of tabs and spaces. Please reformat with spaces only.

Reply:
Done